libraries/ESP8266HTTPClient/src/ESP8266HTTPClient.h

```diff
@@ -124,15 +128,21 @@ typedef enum {
     HTTPC_TE_CHUNKED
 } transferEncoding_t;
 
+#ifdef KEEP_PRESENT_API
 class TransportTraits;
 typedef std::unique_ptr<TransportTraits> TransportTraitsPtr;
+#endif
 
 class HTTPClient
 {
 public:
     HTTPClient();
     ~HTTPClient();
 
+    bool begin(WiFiClient &client, String url);
+    bool begin(WiFiClient &client, String host, uint16_t port, String uri = "/", bool https = false);
+
+#ifdef KEEP_PRESENT_API
```

Comment:
Please mark the functions wrapped in `KEEP_PRESENT_API` block with `__attribute__((deprecated))`

Reply:
Adding   __attribute__ ((deprecated)) to all marked functions. 

@earlephilhower KEEP_PRESENT_API (now changed into HTTPCLIENT_1_1_COMPATIBLE) serves a few goals for me:
- Now that I am working on the library and fully understand it, I am already writing the final, full-rev code. The only thing needed for the full-rev is simply removing everything marked within the #ifdef HTTPCLIENT_1_1_COMPATIBLE ... #endif. I already tested the library without #define HTTPCLIENT_1_1_COMPATIBLE
- If someone wants to compile the library without the #define it is possible, thus saving memory
- By maintaining backward compatibility, it is possible to merge this library into a minor release, but, as already said, it is fully prepared for a major release